packages/errors/README.md

```diff
@@ -46,6 +79,43 @@ OperationalError.isErrorMarkedAsOperational(new OperationalError('example messag
 OperationalError.isErrorMarkedAsOperational(new Error('example message')); // false
 ```
 
+### `HttpError`
+
+The `HttpError` class extends `OperationalError` and represents an HTTP error status. It can work in the same way as a normal error, expecting a message. In this case it will represent an HTTP `500`:
+
+```js
+throw new HttpError('example message');
+```
+
+You can alternatively construct an HTTP error with a data object. This accepts a `statusCode` property, which is a valid HTTP status code number, as well as all of the properties you can set in [`OperationalError`](#operationalerror):
+
+```js
+throw new HttpError({
+    message: 'your thing was not found',
+    statusCode: 404
+});
+```
+
+It's also possible to create an HTTP error with a status code alone, which will default the message to the corresponding HTTP status message:
+
+```js
+throw new HttpError(404);
+```
+
+Error properties can be accessed like any other property:
+
+```js
+error.message // your thing was not found
+error.statusCode // 404
+error.status // 404
+error.statusMessage // Not Found
+error.code // HTTP_404
+```
+
+#### Why use this over `http-errors`?
+
+The benefit of using this error rather than the excellent [http-errors](https://github.com/jshttp/http-errors#readme) library is that we extend `OperationalError` by default. This means that all HTTP errors you throw are considered "known errors" by the rest of our tooling. We also set a `code` property by default which results in less code in our monitoring dashboards – we don't need to check both `code` and `statusCode` properties to determine the type of error thrown.
+
 
 ## Contributing
 
```

Comment:
**praise**: this documentation is fantastic!

**suggestion**: i would love to see more fleshed-out, semi-real-world examples of how and when you'd use each of these error classes in context. with the current examples, i could see somebody doing something like:

```js
const response = await fetch('https://example.com')

if(!response.ok) {
	throw new HttpError({
		message: 'error fetching example.com'
		statusCode: response.statusCode
	})
}
```

and calling it a day, which is obviously not a very helpful error to be throwing. examples of how to use these types effectively (and what _not_ to do!) could be really important.

Reply:
I agree, I think I'd like to get to this once we have a few more modules in place and a real-world implementation. I'd lover for us to produce a go-to document for good practices in error handling. I created an issue (#15) to remind us to do this and will pop something in Jira when we're closer to being ready to write this!